Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyFilePathCtrl.cpp

```diff
@@ -0,0 +1,324 @@
+        // since we enabled it with setClearButtonEnabled.
+        // When it is pressed, it clears the browse edit text for us, but we need to do
+        // some extra logic as well to clear our internal state and trigger our value changed.
+        QToolButton* clearButton = AzQtComponents::LineEdit::getClearButton(m_browseEdit->lineEdit());
+        AZ_Assert(clearButton, "Clear button missing from BrowseEdit");
+        QObject::connect(clearButton, &QToolButton::clicked, this, &PropertyFilePathCtrl::OnClearButtonClicked);
+
+        layout->setContentsMargins(0, 0, 0, 0);
+        layout->addWidget(m_browseEdit);
+
+        setLayout(layout);
+
+        setFocusProxy(m_browseEdit->lineEdit());
+        setFocusPolicy(m_browseEdit->lineEdit()->focusPolicy());
+    };
+
+    void PropertyFilePathCtrl::SetFilePath(const AZ::IO::Path& filePath)
+    {
+        m_currentFilePath = filePath;
+        m_browseEdit->setText(filePath.c_str());
+
+        Q_EMIT FilePathChanged();
+    }
+
+    AZ::IO::Path PropertyFilePathCtrl::GetFilePath() const
+    {
+        return m_currentFilePath;
+    }
+
+    void PropertyFilePathCtrl::SetFilter(const QString& filter)
+    {
+        m_filter = filter;
+    }
+
+    void PropertyFilePathCtrl::SetProductExtension(const AZStd::string& extension)
+    {
+        m_productExtension = extension;
+    }
+
+    void PropertyFilePathCtrl::SetDefaultFileName(const AZStd::string& fileName)
+    {
+        m_defaultFileName = fileName;
+    }
+
+    void PropertyFilePathCtrl::OnOpenButtonClicked()
+    {
+        // Retrieve what the pre-selected file path should be for the file dialog
+        QString preselectedFilePath = GetPreselectedFilePath();
+
+        // Popup a native file dialog to choose a new or existing file
+        // Using the AzQtComponents::FileDialog::GetSaveFileName helper will protect the user from entering
+        // a filename with invalid characters for the AP
+        QString newFilePath =
+            AzQtComponents::FileDialog::GetSaveFileName(this, QObject::tr("Open/Save File"), preselectedFilePath, m_filter);
+
+        // Try to set the new file path (if one was selected)
+        // This will handle file paths outside of the safe asset folders
+        HandleNewFilePathSelected(newFilePath);
+    }
+
+    QString PropertyFilePathCtrl::GetPreselectedFilePath() const
+    {
+        QString preselectedFilePath;
+
+        // If we don't have any output file set yet, then generate a default
+        // option for the user
+        if (m_currentFilePath.empty())
+        {
+            AZ::IO::Path defaultPath = AZ::IO::Path(AZ::Utils::GetProjectPath());
+
+            // If a default filename has been specified, append it to
+            // the project path to be the default path
+            if (!m_defaultFileName.empty())
+            {
+                defaultPath /= AZ::IO::Path(m_defaultFileName);
```

Comment:
You can make the `m_defaultFileName` variable an AZ::IO::Path here to prevent making a copy of it.
Or you can make an `AZ::IO::PathView` out `m_defaultFileName` which makes a string_view backed path and avoids making a copy as well
```suggestion
                defaultPath /= AZ::IO::PathView(m_defaultFileName);
```

Reply:
Thanks yeah I can switch it to `AZ::IO::Path`!